client/src/components/History/CurrentHistory/HistoryOperations/SelectionOperations.vue

```diff
@@ -304,6 +316,21 @@ export default {
         async buildDatasetList() {
             await this.buildNewCollection("list");
         },
+        async buildDatasetListAll() {
+            let allContents = [];
+            const filters = HistoryFilters.getQueryDict(this.filterText);
+
+            allContents = await getHistoryContent(this.history.id, filters, "dataset");
+
+            const modalResult = await buildCollectionModal("list", allContents, this.history.id);
+            await createDatasetCollection(this.history, modalResult);
+
+            // have to hide the source items if that was requested
+            if (modalResult.hide_source_items) {
+                this.$emit("hide-selection", allContents);
+            }
+            this.$emit("reset-selection");
```

Comment:
To reduce duplication, this block could be replaced with something like:
```suggestion
            return this.buildNewCollection("list", allContents);
```
Assuming you modify the `buildNewCollection` function to add a new parameter `contents` that defaults to `this.contentSelection` if unset.